sdl_android/src/main/java/com/smartdevicelink/transport/RouterServiceValidator.java

```diff
@@ -417,16 +417,17 @@ else if(getSecurityLevel(context) == MultiplexTransportConfig.FLAG_MULTI_SECURIT
 		final JSONObject object = new JSONObject();
 		JSONArray array = new JSONArray();
 		JSONObject jsonApp;
-		
-		for(SdlApp app: apps){	//Format all the apps into a JSON object and add it to the JSON array
-			try{
-				jsonApp = new JSONObject();
-				jsonApp.put(JSON_APP_PACKAGE_TAG, app.packageName);
-				jsonApp.put(JSON_APP_VERSION_TAG, app.versionCode);
-				array.put(jsonApp);
-			}catch(JSONException e){
-				e.printStackTrace();
-				continue;
+		if (apps != null && apps.size() > 0) {
```

Comment:
This list should theoretically never be null or size = 0. However, if that is the case I don't believe we should even perform the Http request. We should  add `if(listCallback!=null){listCallback.onListObtained(false);}` and return out of this method with false.